test/v2/issues/271/asyncapi.gen.go

```diff
@@ -0,0 +1,505 @@
+	// Unsubscribing remaining channels
+}
+
+// PublishEvent will publish messages to 'channel' channel
+func (c *UserController) PublishEvent(
+	ctx context.Context,
+	msg ChannelMessage,
+) error {
+	// Get channel path
+	path := "channel"
+
+	// Set context
+	ctx = addUserContextValues(ctx, path)
+	ctx = context.WithValue(ctx, extensions.ContextKeyIsDirection, "publication")
+
+	// Convert to BrokerMessage
+	brokerMsg, err := msg.toBrokerMessage()
+	if err != nil {
+		return err
+	}
+
+	// Set broker message to context
+	ctx = context.WithValue(ctx, extensions.ContextKeyIsBrokerMessage, brokerMsg.String())
+
+	// Publish the message on event-broker through middlewares
+	return c.executeMiddlewares(ctx, &brokerMsg, func(ctx context.Context) error {
+		return c.broker.Publish(ctx, path, brokerMsg)
+	})
+}
+
+// AsyncAPIVersion is the version of the used AsyncAPI document
+const AsyncAPIVersion = "0.0.2"
+
+// controller is the controller that will be used to communicate with the broker
+// It will be used internally by AppController and UserController
+type controller struct {
+	// broker is the broker controller that will be used to communicate
+	broker extensions.BrokerController
+	// subscriptions is a map of all subscriptions
+	subscriptions map[string]extensions.BrokerChannelSubscription
+	// logger is the logger that will be used² to log operations on controller
+	logger extensions.Logger
+	// middlewares are the middlewares that will be executed when sending or
+	// receiving messages
+	middlewares []extensions.Middleware
+	// handler to handle errors from consumers and middlewares
+	errorHandler extensions.ErrorHandler
+}
+
+// ControllerOption is the type of the options that can be passed
+// when creating a new Controller
+type ControllerOption func(controller *controller)
+
+// WithLogger attaches a logger to the controller
+func WithLogger(logger extensions.Logger) ControllerOption {
+	return func(controller *controller) {
+		controller.logger = logger
+	}
+}
+
+// WithMiddlewares attaches middlewares that will be executed when sending or receiving messages
+func WithMiddlewares(middlewares ...extensions.Middleware) ControllerOption {
+	return func(controller *controller) {
+		controller.middlewares = middlewares
+	}
+}
+
+// WithErrorHandler attaches a errorhandler to handle errors from subscriber functions
+func WithErrorHandler(handler extensions.ErrorHandler) ControllerOption {
+	return func(controller *controller) {
+		controller.errorHandler = handler
+	}
+}
+
+type MessageWithCorrelationID interface {
+	CorrelationID() string
+	SetCorrelationID(id string)
+}
+
+type Error struct {
+	Channel string
+	Err     error
+}
+
+func (e *Error) Error() string {
+	return fmt.Sprintf("channel %q: err %v", e.Channel, e.Err)
+}
+
+// ChannelMessage is the message expected for 'ChannelMessage' channel.
+type ChannelMessage struct {
+	// Payload will be inserted in the message payload
+	Payload ChannelMessagePayload
```

Comment:
This is the line that causes the compile error. 

What should actually be used here is the name of the struct that was actually used.
```suggestion
	Payload PayloadSchema
```

Reply:
When downgrading to 0.45.1 the generated file looks exactly the same except that the correct struct name is used as my previous comment